### PR TITLE
Fix/sta queryoptions parsing

### PR DESCRIPTION
**Pull request by SpeckiJ**
Multiple bugfixes related to OData Query Options parsing inside svalbard.
+ code cleanup

### CODE REVIEW

File: shetland/core/src/main/java/org/n52/shetland/oasis/odata/query/option/QueryOptions.java [hunk 1]
```diff
@@ -43,33 +43,33 @@
...
-    private OrderByFilter orderByOption;
-    private SelectFilter selectOption;
-    private ExpandFilter expandOption;
+    private OrderByFilter OrderByFilter;
+    private SelectFilter SelectFilter;
+    private ExpandFilter ExpandFilter;
     private SkipTopFilter skipOption;
     private SkipTopFilter topOption;
-    private FilterFilter filterOption;
+    private FilterFilter FilterFilter;
```
Comment (autermann) [inline shetland/core/src/main/java/org/n52/shetland/oasis/odata/query/option/QueryOptions.java:51]: How about lowercase?

File: svalbard/odata/src/main/antlr4/org/n52/svalbard/odata/grammar/STAQueryOptionsLexer.g4 [hunk 2]
```diff
@@ -279,6 +279,9 @@ EXPAND_AMPERSAND
    : AMPERSAND -> popMode , popMode , type (AMPERSAND)
    ;
 
+mode Dummy;
+DUMMY : 'ä' ;
+
```
Comment (autermann) [inline svalbard/odata/src/main/antlr4/org/n52/svalbard/odata/grammar/STAQueryOptionsLexer.g4:284]: Is this really required?

Comment (SpeckiJ) [reply svalbard/odata/src/main/antlr4/org/n52/svalbard/odata/grammar/STAQueryOptionsLexer.g4]: remove in 1e59db6

File: shetland/core/src/main/java/org/n52/shetland/oasis/odata/query/option/QueryOptions.java (hunk 1 above)
Comment (SpeckiJ) [reply shetland/core/src/main/java/org/n52/shetland/oasis/odata/query/option/QueryOptions.java]: fixed in f017ef1

**Pull request merged**